File: py-test/cpl_flash.py
'''
Created on Jul 10, 2016

@author: Phil
'''

import dirutils
import json2
import utils
import Dir

def doFlashDrive(vl):
    cfn = utils.expES('%FLASH0%\\Projects\\py-test\\plaunch_flash_' + vl + '.json')
    splobj = {}
    try:
        splobj = json2.readConfig(cfn)
    except Exception as e:
        utils.errlog(e)
        raise e

    for k in splobj['paths'].copy().keys():
        if not splobj['paths'][k]['include']:
            del splobj['paths'][k]

    ae1 = dirutils.dirsContainingType(
        Dir.findDL(vl) + '\\',
        {".acm",".ax",".bat",".cmd",".com",".cpl",".dll",".drv",".efi",".exe",".js",".jse",".msc",".mui",".ocx",".scr",".sys",".tsp",".vbe",".vbs",".wsf",".wsh"},
        False,
        [Dir.findDL(vl) + '\\' + 'FQZGJQYA']
        )
    def run(k):
        return k['path'].lower()
    ae2 = sorted(ae1, key=run)
    for a in ae2:
        if a['path'][1:2] == ':':
            a['path'] = a['path'][2:]
    for pd in ae2:
        def ni():
            if not pd['path'] in splobj['paths']:
                return pd['include']
            else:
                return splobj['paths'][pd['path']]['include']
        splobj['paths'][pd['path']] = {
            'include': ni()
            # ,'files': pd.files
        }
    json2.writeConfig(splobj, cfn)

def run():
    # doFlashDrive('CODE')
    doFlashDrive('CODE0')

if __name__ == "__main__":
    run()

File: js-tools/lib/cpl_flash.js
var dirutils = require("dirutils");
var fs = require("fs");
var json = require("json");
var utils = require("utils");
var vol = require("vol");

function doFlashDrive(vl) {
  var cfn = vol.findDL(vl) + "\\Projects\\tools\\plaunch_flash_" + vl + ".json";
  var splobj = {};
  try {
    splobj = json.readConfig(cfn);
  } catch (e) {
    utils.errlog(e);
  }
  for (var k in splobj.paths) {
    if (!splobj.paths[k].include) {
      delete splobj.paths[k];
    }
  }
  return dirutils
    .dirsContainingType(
      vol.findDL(vl) + "\\",
      [
        ".acm",
        ".ax",
        ".bat",
        ".cmd",
        ".com",
        ".cpl",
        ".dll",
        ".drv",
        ".efi",
        ".exe",
        ".js",
        ".jse",
        ".msc",
        ".mui",
        ".ocx",
        ".scr",
        ".sys",
        ".tsp",
        ".vbe",
        ".vbs",
        ".wsf",
        ".wsh",
      ],
      false,
      [vol.findDL(vl) + "\\" + "FQZGJQYA"]
    )
    .then(function (ae1) {
      var ae2 = ae1.sort(function (a, b) {
        return a.path.toLowerCase().localeCompare(b.path.toLowerCase());
      });
      ae2.forEach(function (a) {
        if (a.path.substring(1, 2) == ":") {
          a.path = a.path.substring(2);
        }
      });
      ae2.forEach(function (pd) {
        // add new ones
        var ni = function () {
          if (typeof splobj.paths[pd.path] == "undefined") return pd.include;
          else return splobj.paths[pd.path].include;
        };
        splobj.paths[pd.path] = {
          include: ni(),
          //			,'files': pd.files
        };
      });
      json.writePlaunchJSON(splobj, cfn, true);
    });
}
module.exports.doFlashDrive = doFlashDrive;

function run() {
  // doFlashDrive('CODE');
  return doFlashDrive("CODE0");
}
module.exports.run = run;
